### PR TITLE
Rm transaction_expiry_hr

### PR DESCRIPTION
**Pull request by ghost-not-in-the-shell**
Explain your changes:
This PR removes `transaction_expiry_hr`, which removes the check for transaction expiry in the pool.

Explain how you tested your changes:
*

Checklist:

- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

* Closes #0000

### CODE REVIEW

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

File: src/lib/network_pool/transaction_pool.ml [hunk 1]
```diff
@@ -2272,124 +2267,6 @@ let%test_module _ =
           assert_pool_txs t (List.drop few_now 1) ;
           Deferred.unit )
 
-    let%test_unit "Expired transactions that are already in the pool are \
```
Comment (deepthiskumar) [inline src/lib/network_pool/transaction_pool.ml:2275]: This test never tested the `valid_until` field of a zkApp transaction. Could you please add a test similar to this but testing that expired zkApp transactions (based on the `valid_until` field) are removed from the pool?

**Review by deepthiskumar — CHANGES_REQUESTED** [review]

File: src/lib/mina_base/user_command.ml [hunk 2]
```diff
@@ -231,8 +231,12 @@ let valid_until (t : t) =
   match t with
   | Signed_command x ->
       Signed_command.valid_until x
-  | Zkapp_command _ ->
-      Mina_numbers.Global_slot.max_value
+  | Zkapp_command { fee_payer; _ } -> (
+      match fee_payer.Account_update.Fee_payer.body.valid_until with
```
Comment (deepthiskumar) [inline src/lib/mina_base/user_command.ml:235]: good catch!

**Review by deepthiskumar — APPROVED** [review]

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Pull request merged**